bootstrap-gruntwork-installer.sh

```diff
@@ -193,14 +206,24 @@ only to be installed, but also to execute while a server is booting, and instead
 all over the file system, /etc/user-data gives us a single, common place to put them all.
 EOF
 
-  sudo chown -R "$user_data_folder_owner" "$user_data_folder"
+
+  if [[ "$no_sudo" == "true" ]]; then
+    mkdir -p "$user_data_folder"
+    mv /tmp/gruntwork_installer_user_data_folder_readme.txt "$user_data_folder_readme"
+    chown -R "$user_data_folder_owner" "$user_data_folder"
+  else
+    sudo mkdir -p "$user_data_folder"
+    sudo mv /tmp/gruntwork_installer_user_data_folder_readme.txt "$user_data_folder_readme"
+    sudo chown -R "$user_data_folder_owner" "$user_data_folder"
```

Comment:
The amount of copy/paste and number of places we have to remember to do the `no_sudo` check is a bit worrying.

A thought: if the user sets `--no-sudo`, could we at the start of the script create an `alias` that sets `sudo` to a passthrough? Then the rest of the code in this script would stay the same? Or if `alias` doesn't work, modify `PATH` within this script to have `sudo` be a passthrough script (e.g., one we write to `/tmp`)?

Reply:
This was a fair point and after some experimentation, I was able to define a function that does the work of switching 7da4512